### ISSUE TITLE
Implement DataStore.observe with predicate

### ISSUE THREAD

**Issue by drochetti** [aws-amplify/amplify-android#1322]
Thanks @drochetti , i was trying to observe with a predicate and then filtering with:
```
if(messagesChanged.type().equals(DataStoreItemChange.Type.CREATE)) {
.... update UI
}
```
but it gives this exception first:
DataStoreException{message=Not implemented yet, buster!, cause=null, recoverySuggestion=Check back later!}

do you know when it wil be implemented observe with predicates?

_Originally posted by @marenero1974 in https://github.com/aws-amplify/amplify-android/issues/1310#issuecomment-841536568_

**Comment by richardmcclellan** [comment]
Resolved with the [Amplify Android 1.19.0](https://github.com/aws-amplify/amplify-android/releases/tag/release_v1.19.0) release.

**Comment by marenero1974** [comment]
@richardmcclellan thank you very much
